metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -1156,40 +1157,27 @@ class MetalsLspService(
     } else {
       publishSynthetics(path)
       worksheetProvider.onDidFocus(path)
-      buildTargets.inverseSources(path) match {
-        case Some(target) =>
-          val isAffectedByCurrentCompilation =
-            path.isWorksheet ||
-              buildTargets.isInverseDependency(
-                target,
-                compilations.currentlyCompiling.toList,
-              )
-
-          def isAffectedByLastCompilation: Boolean =
-            !compilations.wasPreviouslyCompiled(target) &&
-              buildTargets.isInverseDependency(
-                target,
-                compilations.previouslyCompiled.toList,
-              )
-
-          val needsCompile =
-            isAffectedByCurrentCompilation || isAffectedByLastCompilation
-          if (needsCompile) {
-            compilations
-              .compileFile(path)
-              .map(_ => DidFocusResult.Compiled)
-              .asJava
-          } else {
-            CompletableFuture.completedFuture(
-              DidFocusResult.AlreadyCompiled
-            )
-          }
-        case None =>
-          CompletableFuture.completedFuture(DidFocusResult.NoBuildTarget)
-      }
+      maybeCompileOnDidFocus(path, prevBuildTarget).asJava
     }
   }
 
+  private def maybeCompileOnDidFocus(
+      path: AbsolutePath,
+      prevBuildTarget: b.BuildTargetIdentifier,
+  ) =
+    buildTargets.inverseSources(path) match {
+      case Some(target) if prevBuildTarget != target =>
+        compilations
+          .compileFile(path)
+          .map(_ => DidFocusResult.Compiled)
+      case _ if path.isWorksheet =>
+        compilations
+          .compileFile(path)
+          .map(_ => DidFocusResult.Compiled)
+      case _ =>
+        Future.successful(DidFocusResult.NoBuildTarget)
```

Comment:
```suggestion
      case Some(_) =>
        Future.successful(DidFocusResult.AlreadyCompiled)
      case None =>
        Future.successful(DidFocusResult.NoBuildTarget)
```
I don't think its used anywhere but maybe for consistency?